### ISSUE TITLE
This could be the beginning of a dream come true...

### ISSUE THREAD

**Issue by jan-jockusch** [Klappson/zope-docker#1]
Strange title for an issue, I know…

This is a nice little docker approach, all-in-one, but I feel we could go much farther with this using a docker-compose approach.

Which would be to use one type of docker container for each component of our system, like so
- One container for one Zope front-end (Zope process and helper tools, mostly from `python-perfact`)
- One container for one ZEO back-end (ZEO process and zodbsync with git repository)
- One container for PostgreSQL (we don't have to do that one ourselves, we can use what's there already)
- One container for HAproxy (should be available, too)
- One YAML file that binds them together with TCP, and exposes the result to the world

Then we could scale up by adding more Zope dockers, which we would then configure into HAproxy. Simple! 

Kubernetes would also become an option…

So this is just me dreaming. If you find this intriguing and would like to explore more, that would make my day!

**Comment by jan-jockusch** [comment]
BTW… see https://git.perfact.de/arolfes/perfact-docker

**Comment by Klappson** [comment]
I already unlocked the compose-sidequest while researching for this approach. This is actually the second time I get a suggestion like this.

However, the target purpose of this implementation is to make creating and deploying small zope-apps more comfortable. This isn't targetting huge enterprise applications. I'm more thinking of apprentices who want to play around with zope or want to use it for school-projects without having to deal with the hussle around it.

BUT the idea of building one of our systems with compose quite tickles my beans to be honest and is something that has left it's mark in my brain. Some tries of that are maybe already sleeping on my disk =)

I can imagine to build both, an all-in-one solution and a more complex compose solution.

I'm gonna keep this issue open and might close it sometime in the future with a link to git.perfact.

Stay tuned.

**Comment by jan-jockusch** [comment]
Yeah, I just wanted to give a little nudge… I'm really quite thrilled to have maybe an alternative to the localdev VM approach.

Plus, since we have Alex in our team, who knows his way around Docker quite well, I thought I'd point you in his direction.

**Comment by Klappson** [comment]
https://github.com/Klappson/zope-compose

:eyes:

**Comment by jan-jockusch** [comment]
So cool! I'll have a look as soon as I can!
